components/site-search-trigger.js

```diff
@@ -0,0 +1,250 @@
+    setRecentTermsJson(limitedRecentTerms);
+  }
+
+  return {
+    // Array of recent terms, most recent first
+    recentTerms,
+    // Called to add a new term to the list of recent terms
+    saveRecentTerm,
+  };
+}
+
+/**
+ * Displays the search trigger for when the navigation is expanded.
+ */
+function SearchTriggerExpanded({ onClick }) {
+  return (
+    <button
+      onClick={onClick}
+      className="flex h-8 w-full items-center gap-2 rounded border border-panel bg-form-element p-2 text-form-element"
+      label="Search the site for a word or phrase"
+    >
+      <MagnifyingGlassIcon className="h-4 w-4" />
+      <div className="text-sm text-gray-400">{`Search (${UC.cmd}K or ${UC.ctrl}K)`}</div>
+    </button>
+  );
+}
+
+SearchTriggerExpanded.propTypes = {
+  // Function to call when the user clicks the search trigger
+  onClick: PropTypes.func.isRequired,
+};
+
+/**
+ * Displays the search trigger for when the navigation is collapsed.
+ */
+function SearchTriggerCollapsed({ onClick }) {
+  return (
+    <button
+      onClick={onClick}
+      className="block"
+      label="Search the site for a word or phrase"
+    >
+      <MagnifyingGlassIcon className="h-8 w-8" />
+    </button>
+  );
+}
+
+SearchTriggerCollapsed.propTypes = {
+  // Function to call when the user clicks the search trigger
+  onClick: PropTypes.func.isRequired,
+};
+
+/**
+ * Displays the site-search modal as an overlay on whatever page the user viewed when they
+ * triggered this box. This also displays the trigger for the box, which appears like a text edit
+ * field though it's actually a button.
+ */
+export default function SiteSearchTrigger({ isExpanded }) {
+  // True if the search input modal is open
+  const [isInputOpen, setIsInputOpen] = useState(false);
+  // Holds the currently typed search term
+  const [searchTerm, setSearchTerm] = useState("");
+  // Recently searched terms from sessionStorage
+  const { recentTerms, saveRecentTerm } = useRecentSearches();
+
+  const inputRef = useRef(null);
+  const router = useRouter();
+
+  // Called to change the search text input value.
+  function onChange(event) {
+    setSearchTerm(event.target.value);
+  }
+
+  // Called to close the search input modal and clear its value for next time.
+  function closeModal() {
+    setSearchTerm("");
+    setIsInputOpen(false);
+  }
+
+  // Navigates to the site-search page with the current search term to show the results. The search
+  // term gets saved to sessionStorage.
+  function closeModalAndExecuteSearch() {
+    closeModal();
+    const trimmedTerm = searchTerm.trim();
+    if (trimmedTerm) {
+      saveRecentTerm(trimmedTerm);
+      router.push(`/site-search?term=${encodeUriElement(trimmedTerm)}`);
+    }
+  }
+
+  // Called when the user types a key while the search input is open, allowing ESC to close the
+  // modal, or RETURN to close the modal and trigger the search.
+  function onKeyDown(event) {
```

Comment:
I'm curious if we also want a button or something in the modal that performs the search also, like pressing Enter does. But I think that's kinda old-school maybe? So maybe we don't care about that.

Reply:
ohhh wait nevermind, the magnifying glass is the search button.